src/operations/extrusions/extrudeFromSlices.js

```diff
@@ -6,13 +6,12 @@ const slice = require('./slice')
 
 const extrudeWalls = require('./extrudeWalls')
 
-// NOTE: function definition is required in order to access 'this'
-function defaultCallback (t, index) {
+const defaultCallback = (percent, index, base) => {
   let baseSlice = null
-  if (geom2.isA(this)) baseSlice = slice.fromSides(geom2.toSides(this))
-  if (poly3.isA(this)) baseSlice = slice.fromPoints(poly3.toPoints(this))
+  if (geom2.isA(base)) baseSlice = slice.fromSides(geom2.toSides(base))
+  if (poly3.isA(base)) baseSlice = slice.fromPoints(poly3.toPoints(base))
 
-  return t === 0 || t === 1 ? slice.transform(mat4.fromTranslation([0, 0, t]), baseSlice) : null
+  return percent === 0 || percent === 1 ? slice.transform(mat4.fromTranslation([0, 0, percent]), baseSlice) : null
```

Comment:
instead of percent, would 'progress' not make more sense ? (can be bulk replaced :D )

Reply:
Done